src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/TimeToLiveService.java

```diff
@@ -27,30 +33,78 @@ public class TimeToLiveService {
+    private final ApplicationParams applicationParams;
+    private final CaseDataService caseDataService;
 
     @Autowired
     public TimeToLiveService(@Qualifier("DefaultObjectMapper") ObjectMapper objectMapper,
-                             ApplicationParams applicationParams) {
+                             ApplicationParams applicationParams,
+                             CaseDataService caseDataService) {
         this.objectMapper = objectMapper;
         this.applicationParams = applicationParams;
+        this.caseDataService = caseDataService;
+    }
+
+    public boolean isCaseTypeUsingTTL(@NonNull CaseTypeDefinition caseTypeDefinition) {
+        return Optional.ofNullable(caseTypeDefinition.getCaseFieldDefinitions()).orElse(Collections.emptyList())
+            .stream().anyMatch(caseFieldDefinition -> TTL_CASE_FIELD_ID.equals(caseFieldDefinition.getId()));
+    }
+
+    public Map<String, JsonNode> updateCaseDataClassificationWithTTL(Map<String, JsonNode> data,
+                                                                     Map<String, JsonNode> dataClassification,
+                                                                     CaseEventDefinition caseEventDefinition,
+                                                                     CaseTypeDefinition caseTypeDefinition) {
+        Map<String, JsonNode> clonedDataClassification = dataClassification;
+        Integer ttlIncrement = caseEventDefinition.getTtlIncrement();
+
+        // if TTL is in play then ensure data classification contains TTL data
+        if (isCaseTypeUsingTTL(caseTypeDefinition) && (ttlIncrement != null)
+            && data != null && data.get(TTL_CASE_FIELD_ID) != null) {
+
+            // using just the TTL data ....
+            Map<String, JsonNode> justTtlData = new HashMap<>();
+            justTtlData.put(TTL_CASE_FIELD_ID, data.get(TTL_CASE_FIELD_ID));
+
+            // ... generate just the TTL data classification
+            Map<String, JsonNode> justTtlDataClassification = caseDataService.getDefaultSecurityClassifications(
+                caseTypeDefinition,
+                justTtlData,
+                new HashMap<>()
+            );
+
+            // .. then clone current data classification and set the TTL classification
+            clonedDataClassification = cloneOrNewJsonMap(dataClassification);
+            clonedDataClassification.put(TTL_CASE_FIELD_ID, justTtlDataClassification.get(TTL_CASE_FIELD_ID));
+        }
+
+        return clonedDataClassification;
     }
 
     public Map<String, JsonNode> updateCaseDetailsWithTTL(Map<String, JsonNode> data,
-                                                          CaseEventDefinition caseEventDefinition) {
+                                                          CaseEventDefinition caseEventDefinition,
+                                                          CaseTypeDefinition caseTypeDefinition) {
         Map<String, JsonNode> clonedData = data;
```

Comment:
Same as L58 above?

Reply:
Resolved in 51e789c546f14817e16520f57e0d91fae3596a8f